Src/WindowsServer/WindowsServer.Shared/Implementation/DataContracts/AzureInstanceComputeMetadata.cs

```diff
@@ -0,0 +1,196 @@
+                    break;
+                case "placementgroupid":
+                    aimsValue = this.PlacementGroupId;
+                    break;
+                case "platformfaultdomain":
+                    aimsValue = this.PlatformFaultDomain;
+                    break;
+                case "platformupdatedomain":
+                    aimsValue = this.PlatformUpdateDomain;
+                    break;
+                case "publisher":
+                    aimsValue = this.Publisher;
+                    break;
+                case "sku":
+                    aimsValue = this.Sku;
+                    break;
+                case "version":
+                    aimsValue = this.Version;
+                    break;
+                case "vmid":
+                    aimsValue = this.VmId;
+                    break;
+                case "vmsize":
+                    aimsValue = this.VmSize;
+                    break;
+                case "subscriptionid":
+                    aimsValue = this.SubscriptionId;
+                    break;
+                case "resourcegroupname":
+                    aimsValue = this.ResourceGroupName;
+                    break;
+                case "tags":
+                    aimsValue = this.Tags;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(string.Format(CultureInfo.InvariantCulture, "No field named '{0}' in AzureInstanceComputeMetadata.", fieldName));
+            }
+
+            if (aimsValue == null)
+            {
+                aimsValue = string.Empty;
+            }
+
+            return aimsValue;
+        }
+
+        /// <summary>
+        /// Because the Azure IMS is on a non-routable IP we need to do some due diligence in our accepting
+        /// values returned from it. This method takes the fieldname and value received for that field, and
+        /// if we can test that value against known limitations of that field we do so here. If the test fails
+        /// we return the empty string, otherwise we return the string given.
+        /// </summary>
+        /// <param name="fieldName">Name of the field acquired from the call to Azure IMS.</param>
+        /// <returns>The value of the field, verified, or the empty string.</returns>
+        internal string VerifyExpectedValue(string fieldName)
+        {
+            string valueToVerify = this.GetValueForField(fieldName);
+
+            string value = string.Empty;
+            bool valueOk = true;
+            TimeSpan regexTimeout = TimeSpan.FromMilliseconds(AzureInstanceComputeMetadata.RegexTimeoutMs);
+
+            if (fieldName.Equals("resourceGroupName", StringComparison.OrdinalIgnoreCase))
+            {
+                valueOk = valueToVerify.Length <= AzureInstanceComputeMetadata.ResourceGroupNameLengthMax;
+                valueOk &= valueToVerify.Length >= AzureInstanceComputeMetadata.ResourceGroupNameLengthMin;
```

Comment:
The technique of keeping a `bool` and masking it (`&=`) with each clause doesn't let the compiler do any short-circuiting. For example, if the first clause fails (the max length check) we still evaluate all the other clauses (including the regexp). Either combine the clauses into a single expression with `&&` or extract it out into a (static?) helper method with early return.

Reply:
Agreed. Changed.